enhancements/update/targeted-update-edge-blocking.md

```diff
@@ -0,0 +1,966 @@
+
+### Enhanced ClusterVersion representation
+
+[The ClusterVersion `status`][api-cluster-version-status] will be extended with a new `conditionalUpdates` property:
+
+```go
+// conditionalUpdates contains the list of updates that may be
+// recommended for this cluster if it meets specific required
+// conditions. Consumers interested in the set of updates that are
+// actually recommended for this cluster should use
+// availableUpdates. This list may be empty if no updates are
+// recommended, if the update service is unavailable, or if an empty
+// or invalid channel has been specified.
+// +optional
+conditionalUpdates []ConditionalUpdate `json:"conditionalUpdates,omitempty"`
+```
+
+The `availableUpdates` documentation will be adjusted to read:
+
+```go
+// availableUpdates contains updates recommended for this
+// cluster. Updates which appear in conditionalUpdates but not in
+// availableUpdates may expose this cluster to known issues. This list
+// may be empty if no updates are recommended, if the update service
+// is unavailable, or if an invalid channel has been specified.
+```
+
+The new ConditionalUpdate type will have the following schema:
+
+```go
+// ConditionalUpdate represents an update which is recommended to some
+// clusters on the version the current cluster is reconciling, but which
+// may not be recommended for the current cluster.
+// +k8s:deepcopy-gen=true
+type ConditionalUpdate struct {
+	// release is the target of the update.
+	// +required
+	release Release `json:"release"`
+
+	// qualifiers represents the range of issues associated with
+	// updating to the target release. The cluster-version
+	// operator will evaluate all entries, and only recommend the
+	// update if there is at least one entry and all entries
+	// recommend the update.
+	// +optional
+	Qualifiers []ConditionalUpdateQualifier `json:"qualifiers,omitempty"`
+
+	// conditions represents the observations of the conditional update's
+	// current status. Known types are:
+	// * Recommended, for whether the update is recommended for the current cluster.
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"`
+}
+```
+
+The new ConditionalUpdateQualifier type will have the following schema:
+
+```go
+// ConditionalUpdateQualifier represents a reason and cluster-state
+// for not recommending a conditional update.
+// +k8s:deepcopy-gen=true
+type ConditionalUpdateQualifier struct {
+	// url contains information about this qualifier.
+	// +optional
+	URL URL `json:"url,omitempty"`
+
+	// reason is the CamelCase reason for not recommending a
+	// conditional update, in the event that qualifiers match the
+	// cluster state.
+	// +optional
+	Reason string `json:"reason,omitempty"`
```

Comment:
is this the logical key of your slice?  If so, how would it be different than a Name?

Reply:
This and the message are named to match the existing condition properties, because the CVO will be inserting these in the `Recommended` condition unless it can decide that the particular risk does not apply to the current cluster state.

Reply:
When the API PR is created, you'll need to select a key for the slice.  I think users more naturally refer to Names.  I'm willing to re-litigate in an API PR.

Reply:
It's not just an openshift/api.  It will also go into graph-data (openshift/cincinnati-graph-data#1056), and be consumed and emitted by openshift/cincinnati.  I prefer `reason` to match the `Recommended` condition, but I'm fine using `name` since you prefer that after hearing my reasoning.  I've pushed 6be15d9 -> e8c0849 to move to `name`.